File: run_box_opener.py
import logging
import time
import os
import requests
from pybip39 import Mnemonic
from dotenv import load_dotenv

load_dotenv()


logging.basicConfig(
    level=logging.INFO,
    format="%(asctime)s %(levelname)s: %(message)s",
    datefmt="%Y-%m-%d %H:%M:%S"
)
DEVNET_API_KEY = os.environ.get("DEVNET_API_KEY", "none")
API_URL = os.environ.get("API_URL", "https://api.erwin.lol")

if DEVNET_API_KEY != "none":
    API_KEY = DEVNET_API_KEY
else:
    API_KEY = os.environ.get("API_KEY")


def submit_guesses():
    passwords = []

    for x in range(0, 50):
        passwords.append(Mnemonic().phrase)

    logging.info("🔑️ Generated %s guesses" % len(passwords))
    logging.info("➡️ Submitting to oracle")

    url = f"{API_URL}/submit_guesses"
    headers = {
        "x-api-key": API_KEY,
        "content-type": "application/json"
    }
    resp = requests.post(
        url,
        json=passwords,
        headers=headers,
        timeout=60
    )

    if resp.status_code == 202:
        logging.info("✅ Guesses accepted")
        return False
    if resp.status_code == 404:
        logging.info(
            f"❌ Guesses rejected ({resp.status_code}): {resp.text}"
        )
        return False
    else:
    if resp.status_code == 502:
        logging.info(
            f"❌ Guesses rejected ({resp.status_code}): Bad Gateway"
        )
        return False
    else:
        logging.info(
            f"❌ Guesses rejected ({resp.status_code}): {resp.text}"
        )
        return True


def do_loop():
    sleep_time = 10
    while True:
        logging.info("⚙️ Generating guesses")
        try:
            rate_limited = submit_guesses()
            if rate_limited:
                sleep_time += 10
            else:
                sleep_time -= 1
        except Exception as err:
            logging.error(f"⚠️ Error occurred: {str(err)}")

        if sleep_time < 10:
            sleep_time = 10

        time.sleep(sleep_time)


if __name__ == "__main__":
    if not API_KEY:
        logging.error(
            "⚠️ API Key not defined, "
            "set API_KEY environment variable"
        )
    else:
        do_loop()

File: run_box_opener.js
const axios = require('axios');
const bip39 = require('bip39');
require('dotenv').config();

const DEVNET_API_KEY = process.env.DEVNET_API_KEY || "none";
const API_URL = process.env.API_URL || "https://api.erwin.lol";
const API_KEY = DEVNET_API_KEY !== "none" ? DEVNET_API_KEY : process.env.API_KEY;

async function submitGuesses() {
    const passwords = [];
    for (let x = 0; x < 50; x++) {
        passwords.push(bip39.generateMnemonic());
    }

    console.info(`🔑️ Generated ${passwords.length} guesses`);
    console.info("➡️ Submitting to oracle");

    const url = `${API_URL}/submit_guesses`;
    const headers = {
        "x-api-key": API_KEY,
        "content-type": "application/json"
    };

    return await axios.post(url, {
        json: passwords,
        headers: headers,
        timeout: 60000
    }).then(resp => {
        if (resp.status === 202) {
            console.info("✅ Guesses accepted");
            return false;
        }
    }).catch(err => {
        if (err.response.status === 401) {
            console.error(`❌ Guesses rejected (${err.response.status}): Unauthorized,Please check your API key`);
            return false;
        } else if (err.response.status === 404) {
            console.error(`❌ Guesses rejected (${err.response.status}): ${err.response.data}`);
            return false;
        } else if (err.response.status === 500) {
            console.error(`❌ Guesses rejected (${err.response.status}): Internal Server Error`);
            return false;
        } else if (err.response.status === 502) {
            console.error(`❌ Guesses rejected (${err.response.status}): Bad Gateway`);
            return false;
        } else if (err.response.status === 530) {
            console.error(`❌ Guesses rejected (${err.response.status}): Argo Tunnel Error`);
            return false;
        } else {
            console.error(`❌ Guesses rejected (${err.response.status}): Unspecified Error`);
            return true;
        }
    });
}

function doLoop() {
    let sleepTime = 10;
    setInterval(() => {
        console.log("⚙️ Generating guesses");
        submitGuesses().then(rateLimited => {
            if (rateLimited) {
                sleepTime += 10;
            } else {
                sleepTime -= 1;
            }
        }).catch(err => {
            console.error(`⚠️ Error occurred: ${err}`);
        });

        if (sleepTime < 10) {
            sleepTime = 10;
        }

        setTimeout(() => {}, sleepTime * 1000);
    }, sleepTime * 1000);
}

if (require.main === module) {
    if (!API_KEY) {
        console.error("⚠️ API Key not defined, set API_KEY environment variable");
    } else {
        doLoop();
    }
}
